build/create_binaries/add_metadata.jl

```diff
@@ -72,4 +76,23 @@ function add_metadata(project_dir, license_file, output_dir, git_repo)
         ```"""
         println(io, version_info)
     end
+
+    # collect lisences of all dependencies
+    ctx = PackageCompiler.create_pkg_context(project_dir)
+    license_dir = joinpath(output_dir, "dep_licenses") 
+    mkpath(license_dir)
+    for (uuid,pkg_entry) in ctx.env.manifest.deps
+	if isnothing(pkg_entry.tree_hash)
+	    # seems as though stdlib packages don't have a tree_sha. as there doesn't seem to be 
+	    # a different way of detecting those, I'm going to assume that is characteristic 
+	    continue 
+	end
+	install_path = Pkg.Operations.find_installed(pkg_entry.name, uuid, pkg_entry.tree_hash)
+	files = readdir(install_path, join=true)
+	license_file_index =findfirst(x-> isfile(x) && !isnothing(match(r"(?i).*/license.*\b$",x)), readdir(install_path, join=true)) 
```

Comment:
Have you considered using `find_license` from https://github.com/ericphanson/LicenseCheck.jl?
Note also you can use your `files` variable instead of repeating `readdir`.

Reply:
Okay, if the maintainer is still active within the community that's less of a concern. I'll leave it up to maintainers to decide whether I should move over to that or if this is okay

Reply:
I see now that LicenseCheck is actually a dependency of PkgToSoftwareBOM, so it is already an indirect dependency.
May also help with being consistent with the SBOM.

https://github.com/Deltares/Wflow.jl/pull/669/files#diff-54a801f523c8f767061bbcd8c8da625a8d8ce3354a1513a7d674badb77fa14e9R1096-R1100